### PR TITLE
W2 - D2 ViewPager

### PR DESCRIPTION
**Pull request by at-hungtran2**
W2 - D2 ViewPager
#9

### CODE REVIEW

File: app/src/at-hungtran/java/asiantech/internship/summer/viewpager/FavouriteFragment.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return view;
+    }
+
+    public void initView(View view) {
+        recyclerView = view.findViewById(R.id.recycler_view);
+        recyclerView.setHasFixedSize(true);
+        LinearLayoutManager layoutManager = new LinearLayoutManager(getActivity());
+        recyclerView.setLayoutManager(layoutManager);
+        mTimelineItems = createTimelineItem();
+        mFavouriteAdapter = new FavouriteAdapter(mTimelineItems, getActivity());
```
Comment (at-dev-support-01) [inline app/src/at-hungtran/java/asiantech/internship/summer/viewpager/FavouriteFragment.java]: ERROR: [STYLE][ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD] Write to static field asiantech.internship.summer.viewpager.FavouriteFragment.mFavouriteAdapter from instance method asiantech.internship.summer.viewpager.FavouriteFragment.initView(View)
http://findbugs.sourceforge.net/bugDescriptions.html#ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD

File: app/src/at-hungtran/java/asiantech/internship/summer/viewpager/PagerActivity.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import asiantech.internship.summer.R;
+import asiantech.internship.summer.model.TimelineItem;
+
+@SuppressLint("Registered")
+public class PagerActivity extends AppCompatActivity {
+    public static List<TimelineItem> itemList = new ArrayList<>();
```
Comment (at-dev-support-01) [inline app/src/at-hungtran/java/asiantech/internship/summer/viewpager/PagerActivity.java]: ERROR: [MALICIOUS_CODE][MS_SHOULD_BE_FINAL] asiantech.internship.summer.viewpager.PagerActivity.itemList isn't final but should be
http://findbugs.sourceforge.net/bugDescriptions.html#MS_SHOULD_BE_FINAL

File: app/src/at-hungtran/java/asiantech/internship/summer/viewpager/PagerActivity.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import android.annotation.SuppressLint;
+import android.os.Bundle;
+import android.support.design.widget.TabLayout;
+import android.support.v4.view.ViewPager;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+
+import java.util.ArrayList;
+import java.util.Collections;
```
Comment (at-dev-support-01) [inline app/src/at-hungtran/java/asiantech/internship/summer/viewpager/PagerActivity.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.Collections'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-hungtran/java/asiantech/internship/summer/viewpager/TimelineFragment.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ProgressBar;
+
+import java.util.ArrayList;
+import java.util.Collections;
```
Comment (at-dev-support-01) [inline app/src/at-hungtran/java/asiantech/internship/summer/viewpager/TimelineFragment.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.Collections'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/at-hungtran/java/asiantech/internship/summer/viewpager/FavouriteAdapter.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
+package asiantech.internship.summer.viewpager;
+
+import android.content.Context;
```
Comment (at-dev-support-01) [inline app/src/at-hungtran/java/asiantech/internship/summer/viewpager/FavouriteAdapter.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.content.Context'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

Comment (at-thangtruong) [reply app/src/at-hungtran/java/asiantech/internship/summer/viewpager/FavouriteAdapter.java]: Em xoá import thừa đi nghe.

File: app/src/at-hungtran/java/asiantech/internship/summer/viewpager/TimelineFragment.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+import android.widget.ProgressBar;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import asiantech.internship.summer.R;
+import asiantech.internship.summer.model.TimelineItem;
+
+public class TimelineFragment extends Fragment implements TimelineAdapter.onClick {
```
Comment (at-thangtruong) [inline app/src/at-hungtran/java/asiantech/internship/summer/viewpager/TimelineFragment.java]: Em merge code từ nhánh `master` vào chưa? Vì bài tập cũ có class `TimelineFragment` này rồi, em chỉ việc dùng lại thôi.

Comment (at-hungtran2) [reply app/src/at-hungtran/java/asiantech/internship/summer/viewpager/TimelineFragment.java]: @at-thangtruong  em merge master rồi anh, xong em copy qua 1 package mới

Comment (at-thangtruong) [reply app/src/at-hungtran/java/asiantech/internship/summer/viewpager/TimelineFragment.java]: Okay em

**Pull request merged**